oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/query/IndexName.java

```diff
@@ -33,16 +33,8 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * An index name, which possibly contains two version numbers: the product
- * version number, and the customer version number.
- *
- * The format of an index node name is:
- * - The name of the index,
- * - optionally a dash ('-') and the product version number,
- * - optionally "-custom-" and the customer version number.
- *
- * If the node name doesn't contain version numbers / dashes, then version 0 is
- * assumed (for both the product version number and customer version number).
+ * @deprecated
+ * Use oak-core org.apache.jackrabbit.oak.plugins.index.IndexName instead.
```

Comment:
Is it enough to have `@deprecated` in the comments, without having the `@Deprecated` annotation on the class?

Reply:
I think it doesn't matter: the compiler and IDE does see the deprecation. That's probably due to the history of Java: the annotation was added later. But I can add the annotation as well, no issue.